src/rgw/rgw_usage_perf.h

```diff
@@ -0,0 +1,133 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <shared_mutex>
+#include <thread>
+#include <atomic>
+#include <chrono>
+
+#include "common/perf_counters.h"
+#include "rgw_usage_cache.h"
+
+// To avoid heavy header rgw_sal.h , we are adding a forward declaration here
+namespace rgw::sal {
+  class Driver;
+  class Bucket;
+  class User;
+}
+
+namespace rgw {
+
+// Performance counter indices
+enum {
+  l_rgw_usage_first = 920000,
+  l_rgw_usage_cache_hit,
+  l_rgw_usage_cache_miss,
+  l_rgw_usage_cache_update,
+  l_rgw_usage_cache_evict,
+  l_rgw_usage_last
+};
+
+class UsagePerfCounters {
```

Comment:
i was expecting to see [PerfCountersCache](https://github.com/ceph/ceph/blob/f4e964eea6/src/common/perf_counters_cache.h) used for this. is there a reason you couldn't reuse that, or were you just not aware of it?

Reply:
> PerfCountersCache uses LRU eviction, but for usage metrics we need counters to persist as long as the user/bucket exists (not evict based on access patterns)

i really don't think we need to persist counters for all users and buckets in the cluster. we only need a given bucket's counters in memory long enough to send it out to ceph-exporter/prometheus. in the previous design for [per-user and per-bucket op counters](https://github.com/ceph/ceph/pull/53003), we relied on the PerfCountersCache lru to bound the total number of counters we keep in memory